apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs

```diff
@@ -268,7 +268,14 @@ internal V1.Type ToProtobufType()
         /// </summary>
         /// <param name="size"></param>
         /// <returns></returns>
-        public SpannerDbType WithSize(int size) => new SpannerDbType(TypeCode, size);
+        public SpannerDbType WithSize(int size)
+        {
+            if (size != 0 && TypeCode != TypeCode.Bytes && TypeCode != TypeCode.String)
```

Comment:
Should this check be size > 0?

Reply:
Or possibly check that `size >= 0` first, and then check that if it *is* non-zero, that it's for a suitable type?

Reply:
I'll need to double check this one because a negative size (-1) might be used as a marker.  This check should be in the property setter.  it can be here too, but the main place would be there.

Reply:
ok it looks like -1 is only used as a marker internally in some implementations of parameter -- so its ok to disallow on our side.
i've widened this check to disallow calls at all to "WithSize" if the typecode isnt "bytes" or "string" and have added a guard to the ctor when size is set to ensure it's not negative.